File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/presentation/settings/step/SetTimerRepeatFragment.java
package com.igorganapolsky.vibratingwatchapp.presentation.settings.step;

import androidx.lifecycle.ViewModelProviders;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.wear.widget.WearableRecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import com.igorganapolsky.vibratingwatchapp.R;
import com.igorganapolsky.vibratingwatchapp.util.RecyclerViewSnapLayoutManager;
import com.igorganapolsky.vibratingwatchapp.presentation.ViewModelFactory;
import com.igorganapolsky.vibratingwatchapp.presentation.settings.SetTimerViewModel;
import com.igorganapolsky.vibratingwatchapp.presentation.settings.adapter.HolderClickListener;
import com.igorganapolsky.vibratingwatchapp.presentation.settings.adapter.RepeatsAdapter;

public class SetTimerRepeatFragment extends Fragment implements HolderClickListener {

    private WearableRecyclerView wrvRepeats;
    private SetTimerViewModel mViewModel;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mViewModel = ViewModelProviders.of(requireActivity(), ViewModelFactory.getInstance()).get(SetTimerViewModel.class);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.set_timer_repeat_fragment, container, false);
        setupView(rootView);
        return rootView;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        setupObservers();
    }

    private void setupView(View rootView) {
        wrvRepeats = rootView.findViewById(R.id.wrvRepeats);

        RepeatsAdapter adapter = new RepeatsAdapter(this);
        RecyclerViewSnapLayoutManager layoutManager = new RecyclerViewSnapLayoutManager(getContext(), LinearLayoutManager.HORIZONTAL);
        layoutManager.setItemSelectListener((int pos) -> mViewModel.setTimerRepeat(pos));

        wrvRepeats.setAdapter(adapter);
        wrvRepeats.setLayoutManager(layoutManager);
    }

    private void setupObservers() {
        mViewModel.getTimerData().observe(this, (timerValue) -> {
            if (timerValue == null) return;
            wrvRepeats.smoothScrollToPosition(mViewModel.getRepeatPosition());
        });
    }

    @Override
    public void onHolderClick(int position) {
        wrvRepeats.smoothScrollToPosition(position);
    }
}

File: app/src/main/kotlin/com/igorganapolsky/vibratingwatchapp/presentation/settings_screen/step/SetTimerRepeatFragment.kt
package com.igorganapolsky.vibratingwatchapp.presentation.settings_screen.step

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.lifecycle.observe
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.wear.widget.WearableRecyclerView
import com.igorganapolsky.vibratingwatchapp.R
import com.igorganapolsky.vibratingwatchapp.presentation.home_screen.TimerViewModel
import com.igorganapolsky.vibratingwatchapp.presentation.settings_screen.adapter.HolderClickListener
import com.igorganapolsky.vibratingwatchapp.presentation.settings_screen.adapter.RepeatsAdapter
import com.igorganapolsky.vibratingwatchapp.presentation.RecyclerViewSnapLayoutManager

class SetTimerRepeatFragment : Fragment(), HolderClickListener {

    private var wrvRepeats: WearableRecyclerView? = null
    //    private var mViewModel: TimerViewModel? = null
    private val mViewModel by viewModels<TimerViewModel>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
//        mViewModel = ViewModelProviders.of(requireActivity(), ViewModelFactory.instance)
//            .get(TimerViewModel::class.java)
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val rootView = inflater.inflate(R.layout.set_timer_repeat_fragment, container, false)
        setupView(rootView)
        return rootView
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setupObservers()
    }

    private fun setupView(rootView: View) {
        wrvRepeats = rootView.findViewById(R.id.wrvRepeats)

        val adapter = RepeatsAdapter(this)
        val layoutManager =
            RecyclerViewSnapLayoutManager(
                context,
                LinearLayoutManager.HORIZONTAL
            )
        layoutManager.setItemSelectListener { pos: Int -> mViewModel.setTimerRepeat(pos) }

        wrvRepeats!!.adapter = adapter
        wrvRepeats!!.layoutManager = layoutManager
    }

    private fun setupObservers() {
        mViewModel.getTimerData()
            .observe(this) { wrvRepeats!!.smoothScrollToPosition(mViewModel.repeatPosition) }
    }

    override fun onHolderClick(position: Int) {
        wrvRepeats!!.smoothScrollToPosition(position)
    }
}
